orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -116,8 +103,11 @@ class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
     targetPercentages.forEach({ p ->
       def resizeContext = baseContext + [
           target: TargetServerGroup.Params.Target.current_asg_dynamic,
+          action: ResizeStrategy.ResizeAction.scale_to_server_group,
+          source: getSource(stageData, baseContext),
           targetLocation: cleanupConfig.location,
-          capacity: makeIncrementalCapacity(originalCapacity, p)
+          scalePct: p,
+          pinCapacity: p < 100 // if p = 100, capacity should be unpinned
```

Comment:
What does it mean to pin the capacity?

Reply:
min = max = desired

`scale_to_server_group` already supported pinning min = desired, this just builds on that to min, max and desired.